### ISSUE TITLE
Default values for number of replicas and maxUnavailable causes deployments to report as successful rolled out when no pods available

### ISSUE THREAD

**Issue by vieira** [kubernetes/kubernetes#46575]
**CHANGE REQUEST**

**Kubernetes version** (use `kubectl version`):
Client Version: v1.6.4
Server Version: v1.6.2
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T18:44:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: N/A
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04.2 LTS
- **Kernel** (e.g. `uname -a`): Linux 4.8.0-52-generic 55~16.04.1-Ubuntu
- **Install tools**: N/A
- **Others**: N/A

**What happened**:
Default values for number of replicas and maxUnavailable causes deployments to report as successful rolled out when no pods are available.

**What you expected to happen**:
Expected saner defaults that would not report the deployment as successful rolled out when no replicas are available.

**How to reproduce it** (as minimally and precisely as possible):
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: review-test-master
spec:
  template:
    spec:
      containers:
      - name: somename
        image: $CI_REGISTRY_IMAGE
        imagePullPolicy: Always
        ports:
        - containerPort: 8080
      imagePullSecrets:
        - name: $CI_REGISTRY
```
Make this fail by, e.g., specifying some private registry with wrong credentials. Deployment will be successful

```
$ kubectl rollout status deployment/review-test-master --namespace ci
deployment "review-test-master" successfully rolled out
```
although nothing is available:
```
$ kubectl get deploy/review-test-master --namespace ci 
NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
review-test-master   1         1         1            0           9m
```
**Anything else we need to know**:
This is caused by the fact that by default replicas is set to 1 while maxUnavailable is also set to 1 allowing deployments without any running pods to report as successful when default settings are used.

**Comment by janetkuo** [comment]
This is fixed in 1.7 by comparing available replicas with up-to-date replicas instead https://github.com/kubernetes/kubernetes/blob/v1.7.0-alpha.4/pkg/kubectl/rollout_status.go#L80-L82

If you don't want to wait until 1.7 is release, use `apps/v1beta1` Deployment instead -- with 25% default maxUnavailable.

**Comment by vieira** [comment]
Using `apps/v1beta1` instead of `extensions/v1beta1` solves the issue for me. Thanks!
